### DISCUSSION TITLE
Allow for footer items which contain all three of `text`, `icon`, and `href` subkeys (nav-style)

### DISCUSSION THREAD

**Discussion by connortwiegand** [quarto-dev/quarto-cli#10288] [Feature Requests]
### Description

Hi,

In navbars and sidebar `contents`, you can pass an item of the form

```
- icon: foo
  text: bar 
  href: baz.biz
``` 

It would be nice if this was supported in the footer as well. Currently, the `page-footer` key under `website` allows for 
```
- icon: foo
  href: baz.biz
```
**or**
```
- text: foo
  href: baz.biz
```
But not all three. A current workaround is to use, e.g., 
```
- text: "{{< fa rss >}} Feed"
  href: "index.xml"
```
I think that unification to the `text`/`icon`/`href` schema would be clear + useful, if it's not too much trouble to implement. 

Thank you!

**Comment by mcanouil** [comment]
- https://github.com/quarto-dev/quarto-cli/issues/10306
